src/test/java/net/sf/jabref/model/entry/AuthorListTest.java

```diff
@@ -580,4 +580,29 @@ public void testRemoveStartAndEndBraces() {
     public void createCorrectInitials() {
         Assert.assertEquals("J. G.", AuthorList.parse("Hornberg, Johann Gottfried").getAuthor(0).getFirstAbbr());
     }
+
+    @Test
+    public void parseNameWithBracesAroundFirstName() throws Exception {
+        //TODO: Be more intelligent and abbreviate the first name correctly
+        Author expected = new Author("Tse-tung", "{Tse-tung}.", null, "Mao", null);
+        Assert.assertEquals(new AuthorList(expected), AuthorList.parse("{Tse-tung} Mao"));
+    }
+
+    @Test
+    public void parseNameWithBracesAroundLastName() throws Exception {
+        Author expected = new Author("Hans", "H.", null, "van den Bergen", null);
+        Assert.assertEquals(new AuthorList(expected), AuthorList.parse("{van den Bergen}, Hans"));
+    }
+
+    @Test
+    public void parseNameWithHyphenInFirstName() throws Exception {
+        Author expected = new Author("Tse-tung", "T.-t.", null, "Mao", null);
```

Comment:
I think it depends a bit on the convention / citation style which abbreviation is "correct". In the end I decided to leave it like that because it is not a big deal.

@mlep thanks for reporting this issue in the first place. I think the author formatting has still some problems with not-so-common-cases.